tls/ttls.h

```diff
@@ -560,6 +565,8 @@ typedef struct ttls_context {
 	unsigned int		nb_zero;
 	int			client_auth;
 	char			*hostname;
+	char			*sni;
+	char			sni_buf[64];
```

Comment:
These could be union at least (we can determine whether we need to free SNI by `sni_len`)